### ISSUE TITLE
|synced| CRITICAL : [TWISTED] Unhandled Error,   and one more

### ISSUE THREAD

**Issue by converghub** [theQRL/QRL#333]
I got this error, and after a while got another one.

The first one:
```
2017-10-07 16:54:38,137 |synced| CRITICAL : [TWISTED] Unhandled Error
Traceback (most recent call last):
  File "start_qrl.py", line 13, in <module>
    main()
  File "/home/pi/QRL/qrl/main.py", line 136, in main
    reactor.run()
  File "/home/pi/.local/lib/python3.5/site-packages/twisted/internet/base.py", line 1243, in run
    self.mainLoop()
  File "/home/pi/.local/lib/python3.5/site-packages/twisted/internet/base.py", line 1252, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/home/pi/.local/lib/python3.5/site-packages/twisted/internet/base.py", line 878, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "/home/pi/QRL/qrl/core/node.py", line 519, in post_block_logic
    self.create_next_block(blocknumber)
  File "/home/pi/QRL/qrl/core/node.py", line 473, in create_next_block
    blocknumber - 1)
  File "/home/pi/QRL/qrl/core/node.py", line 284, in create_new_block
    block_obj = self.chain.create_stake_block(reveal_hash, vote_hash, last_block_number)
  File "/home/pi/QRL/qrl/core/chain.py", line 272, in create_stake_block
    block_obj = self.m_create_block(reveal_hash, vote_hash, last_block_number)
  File "/home/pi/QRL/qrl/core/chain.py", line 498, in m_create_block
    self.wallet.save_slave(slave_xmss)
  File "/home/pi/QRL/qrl/core/wallet.py", line 60, in save_slave
    outfile.write(w.SerializeToString())
builtins.TypeError: write() argument must be str, not bytes
```

The second one:
```
2017-10-07 23:37:25,646 |syncing| INFO : <<<Fetching block: 14543 from 144.217.5.9:9000
2017-10-07 23:37:26,901 |syncing| INFO : >>> Received Block #14543
2017-10-07 23:36:06,606 |syncing| ERROR : Traceback (most recent call last):
  File "/home/pi/QRL/qrl/core/block.py", line 90, in validate_block
    if not self.blockheader.validate(last_block.blockheader):
AttributeError: 'list' object has no attribute 'blockheader'

2017-10-07 23:36:06,609 |syncing| INFO : m_add_block failed - block failed validation.
2017-10-07 23:36:06,611 |syncing| INFO : Failed to add block by m_add_block, re-requesting the block #14543
```

Here is the log file.
https://transfer.sh/eAXmF/qrl.log

**Comment by jleni** [comment]
The first issue is fixed by PR #335

**Comment by jleni** [comment]
@cyyber You can take over the second one here

**Comment by cyyber** [comment]
Second issue has been fixed by PR #344
